packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -747,6 +747,28 @@ mixin WidgetInspectorService {
   /// The current [WidgetInspectorService].
   static WidgetInspectorService get instance => _instance;
   static WidgetInspectorService _instance = _WidgetInspectorService();
+
+  /// Whether the inspector is in select mode.
+  ///
+  /// In select mode, pointer interactions trigger widget selection instead of
+  /// normal interactions. Otherwise the previously selected widget is
+  /// highlighted but the application can be interacted with normally.
+  @visibleForTesting
+  final ValueNotifier<bool> isSelectMode = ValueNotifier<bool>(true);
+
+  /// For use by [WidgetInspector] widgets so they can be notified when the
+  /// selection has changed.
+  ValueListenable<int> get _selectionChanged => _selectionChangeCount;
```

Comment:
There is no specific need for the count, we are just leveraging the ability to trigger the listeners each time the count changes.

If there are more direct ways to do this I'd be open to them. The other methods I found seemed move convoluted though.

Reply:
It would require some slight refactoring but the most idiomatic would be to use the InspectorSelection object as the value. You would need to tweak it so it is immutable changing the methods that mutate it to return a new InspectorSelectionObject. That would also make it harder to introduce bugs in the future due to the selection change becoming out of sync with calls to _notifiySelectionChanged.

Reply:
@jacob314 I've added these changes to try to simplify the callback triggering cycle.

Since the WidgetInspectorService is a mixing I wasn't able to set the listener directly there, so in the tests I had to duplicate it. I'm not too excited about it, do you know of a better way that listener could come into existence?

Reply:
Oops looks like I didn't post the link, here you go:
https://github.com/flutter/flutter/pull/129452/commits/7748464ceabf1bf486c9308f6d45e018562ec33b